core/domain/state_domain.py

```diff
@@ -737,13 +737,14 @@ def validate(self):
             raise utils.ValidationError(
                 'Expected needs_update to be a bool, received %s' %
                 self.needs_update)
-        if not isinstance(self.duration_secs, float):
+        if not isinstance(self.duration_secs, (float, int)):
```

Comment:
Actually, keep it as float. Normalize to float in the domain-layer method that updates the value in the backend domain object.

Reply:
So under the Voiceover domain class, should I do this under init() method or a separate method?

Reply:
I was thinking more in terms of whatever method is updating the voiceover. I.e. you would normalize to float prior to calling `__init__`.

But I also see no harm in adding it in `__init__` for safety too. That's just a backup option though. We should update the methods that are updating the voiceover data as a first priority since there is an explicit conversion needed from int to float in some cases.

Reply:
Ok I see. Yeah I saw 2 ways of completing this, the way I mentioned earlier in via the Voiceover domain class. The other is to find every submission request (POST/PUT) or data saving methods like you mentioned.

Reply:
Yeah, I would go with the data saving methods -- I mean, somewhere, something must be calling Voiceover(...), right? The normalization should be done to the arguments to that call, prior to the call being made.

(I think it also makes sense to add an extra normalization in `__init__` for safety, but if that gets triggered maybe log an error or something to the backend since it shouldn't occur in practice. Tests should check that.)

Reply:
Handled this by directly converting to float in the backend itself. So done.